.circleci/config.yml

```diff
@@ -0,0 +1,116 @@
+            npm run lint
+            # JS tests for dependencies installed with npm3
+            npm run flow -- check
+            npm test -- --maxWorkers=1
+
+            # build app
+            buck build ReactAndroid/src/main/java/com/facebook/react
+            buck build ReactAndroid/src/main/java/com/facebook/react/shell
+
+            # compile native libs with Gradle script, we need bridge for unit and
+            # integration tests
+            ./gradlew :ReactAndroid:packageReactNdkLibsForBuck -Pjobs=1 -Pcom.android.build.threadPoolSize=1:
+                timeout: 360
+
+            # unit tests
+            buck test ReactAndroid/src/test/... --config build.threads=1
+
+            # integration tests
+            # build JS bundle for instrumentation tests
+            node local-cli/cli.js bundle --max-workers 1 --platform android --dev true --entry-file ReactAndroid/src/androidTest/js/TestBundle.js --bundle-output ReactAndroid/src/androidTest/assets/AndroidTestBundle.js
+            # build test APK
+            buck install ReactAndroid/src/androidTest/buck-runner:instrumentation-tests --config build.threads=1
+            # run installed apk with tests
+            # node ./scripts/run-android-ci-instrumentation-tests.js --retries 3 --path ./ReactAndroid/src/androidTest/java/com/facebook/react/tests --package com.facebook.react.tests
+
+            # Android e2e test
+            # disabled pending on https://our.intern.facebook.com/intern/tasks?t=16912142
+            # source scripts/circle-ci-android-setup.sh && retry3 node ./scripts/run-ci-e2e-tests.js --android --js --retries 2
+
+            # testing docs generation
+            cd website && npm test
+            cd website && node ./server/generate.js
+            # if website can be built, deploy regardless of test failures
+            if [[ ($CIRCLE_BRANCH =~ .*-stable || $CIRCLE_BRANCH == "master") && $CIRCLE_PROJECT_USERNAME == "facebook" && -z $CI_PULL_REQUEST && -z $CIRCLE_PR_USERNAME ]]; then git config --global user.email "reactjs-bot@users.noreply.github.com" && git config --global user.name "Website Deployment Script" && echo "machine github.com login reactjs-bot password $GITHUB_TOKEN" > ~/.netrc && cd website && GIT_USER=reactjs-bot npm run gh-pages; fi
+
+      - run:
+          command: |
+            # copy test report for Circle CI to display
+            mkdir -p $CIRCLE_TEST_REPORTS/junit/
+            find . -type f -regex ".*/build/test-results/debug/.*xml" -exec cp {} $CIRCLE_TEST_REPORTS/junit/ \;
+            find . -type f -regex ".*/outputs/androidTest-results/connected/.*xml" -exec cp {} $CIRCLE_TEST_REPORTS/junit/ \;
+            # TODO circle does not understand Buck's report, maybe need to transform xml slightly
+            #find . -type f -regex ".*/buck-out/gen/ReactAndroid/src/test/.*/.*xml" -exec cp {} $CIRCLE_TEST_REPORTS/junit/ \;
+
+      - deploy:
+          branch: [/.*-stable/, /master/]
```

Comment:
I'm not sure if this syntax is supported anymore since the CircleCI docs say to check the branch name in the `command` section manually.